include/solvers/heat_transfer_scratch_data.h

```diff
@@ -341,6 +345,29 @@ class HeatTransferScratchData
       current_solution, velocity_gradient_values);
   }
 
+  /**
+   *@brief Reinitialize cell's pressure value with value calculated in the Fluid
+   * Dynamics (FD) physic.
+   *
+   * @tparam VectorType Type of vector used for storing fluid dynamics
+   * solutions.
+   *
+   * @param[in] cell The cell for which the pressure is reinitialized
+   * This cell must be compatible with the FD FiniteElement.
+   *
+   * @param[in] current_solution Solution vector from FD containing velocity
+   * components and pressure values.
+   */
+  template <typename VectorType>
+  void
+  reinit_pressure(const typename DoFHandler<dim>::active_cell_iterator &cell,
+                  const VectorType &current_solution)
+  {
+    this->fe_values_fd.reinit(cell);
+    this->fe_values_fd[pressure].get_function_values(current_solution,
+                                                     pressure_values);
+  }
```

Comment:
Wait why is this a second FE_values reinit? This is very expensive as a call. It would be better to gather pressure when we gather the fluid dynamics instead of calling reinit when we need both...